micro_sam/sam_annotator/_widgets.py

```diff
@@ -0,0 +1,56 @@
+from enum import Enum
+from pathlib import Path
+from typing import TYPE_CHECKING, Optional
+
+from magicgui import magic_factory
+from magicgui.tqdm import tqdm
+from superqt.utils import thread_worker
+
+from micro_sam.util import (
+    ImageEmbeddings,
+    get_sam_model,
+    precompute_image_embeddings,
+    _MODEL_URLS,
+    _DEFAULT_MODEL,
+)
+
+if TYPE_CHECKING:
+    import napari
+
+Model = Enum("Model", _MODEL_URLS)
+
+
+@magic_factory(call_button="Compute image embeddings",
+               device = {"choices": ["auto", 'cuda', 'mps', 'cpu']})
+def embedding_widget(
+    image: "napari.layers.Image",
+    model: Model = Model.__getitem__(_DEFAULT_MODEL),
+    device = "auto",
+    save_path: Optional[Path] = None,  # where embeddings for this image are cached (optional)
```

Comment:
Are you sure you want "None" to be the default save path?
I could use something like `pooch.os_cache_dir("micro-sam")/"embeddings"` and use the `image.name` parameter to name the zarr files.

Reply:
Also, this widget is giving us a file picker - I think we want a directory picker here instead.

Reply:
>> Also, this widget is giving us a file picker - I think we want a directory picker here instead.
>
> Yep, that would be better. Do you know how to do that or should we ping Talley?

Yes, I have made it a directory picker. Just had to add `save_path={"mode": "d"},  # choose a directory` to the `magic_factory` decorator, and it's working now.

Reply:
>> Are you sure you want "None" to be the default save path?
>> I could use something like pooch.os_cache_dir("micro-sam")/"embeddings" and use the image.name parameter to name the zarr files.
>
> I am not so sure about this. I am a bit afraid we will take up quite a lot of disc space. I will check how big the embeddings actually get and we can revisit this before merging.

That is a reasonable concern. 
I think I agree with you about leaving it `None`. But that does make it extra-important that the EMBEDDINGS global variable is being set correctly from the thread worker - I'm not totally sure how to check that. Any ideas?